crates/types/src/joints.rs

```diff
@@ -671,6 +671,20 @@ impl Joints<f32> {
         }
     }
 
+    pub fn from_iterator<I>(angles: I) -> Self
+    where
+        I: Iterator<Item = f32>,
+    {
+        let angles_array: [f32; 26] =
+            angles
+                .collect::<Vec<f32>>()
+                .try_into()
+                .unwrap_or_else(|v: Vec<f32>| {
+                    panic!("Expected 26 joints but found {} values", v.len())
+                });
+        Joints::from_angles(angles_array)
```

Comment:
Isn't there a better way to create Joints from an iterator instead of copying it first to an array and then into the Joints?

I thought, you could just construct a new Joints struct and in each field you would call `.next()`.

Another thing to consider: Maybe also implement `FromIterator` for testing the composed structs and then delegating the construction to them from the upper Joints.

Yet another idea: We could also probably delete `Joints::from_angles()` because you can probably just use the new `FromIterator` everywhere.

Reply:
Done.